main/src/ca/uwaterloo/flix/tools/pkg/MavenPackageManager.scala

```diff
@@ -31,27 +33,54 @@ object MavenPackageManager {
     * Installs all MavenDependencies for a Manifest,
     * including transitive dependencies using coursier.
     */
-  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[Unit, PackageError] = {
+  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[List[Path], PackageError] = {
     val depStrings = getMavenDependencyStrings(manifest)
 
     Result.sequence(depStrings.map(createCoursierDependencies)).flatMap { deps =>
-      try {
+      val l = try {
         val res = deps.foldLeft(Resolve())((res, dep) => res.addDependencies(dep))
         val resolution = res.run()
 
-        val fetch = resolution.dependencies.foldLeft(Fetch())((f, dep) => {
-          out.println(s"Installing ${dep.module.toString()}");
-          f.addDependencies(dep)
+        val resList: collection.mutable.ListBuffer[Path] = collection.mutable.ListBuffer.empty
+        val fetch = resolution.dependencies.foldLeft(Fetch())(
+          (f, dep) => {
+            val moduleName = dep.module.toString()
+            val moduleNamePath = moduleName.replaceAll("[^a-zA-Z0-9-]", "/")
+            val versionString = dep.version
+            val fileName = s"${moduleNamePath.split('/').last}-$versionString.jar"
+            val filePrefix = getFilePrefix
+            val path = Paths.get(s"$filePrefix/$moduleNamePath/$versionString/$fileName")
+            resList.addOne(path)
+
+            out.println(s"Installing $moduleName")
+            f.addDependencies(dep)
         })
         fetch.run()
+        resList.toList
       } catch {
         case e: Exception =>
-          out.println(e.getMessage)
+          out.println(e)
           //Shortens the error message to just give the name of the dependency
           val message = e.getMessage.replaceAll("[^a-zA-Z0-9:. ]", "/").split('/').apply(0)
           return Err(PackageError.CoursierError(s"Error in downloading Maven dependency: $message"))
       }
-      ().toOk
+      l.toOk
+    }
+  }
+
+  /**
+    * Gives the correct path prefix for the
+    * Coursier cache based on operating system.
+    */
+  private def getFilePrefix: String = {
```

Comment:
Might be useful to look at https://get-coursier.io/docs/cache

Reply:
Or maybe there is a better way to get the paths than to construct them ourselves.